### PR TITLE
Upgraded nextcloud to version 20.0.1

### PR DESCRIPTION
**Pull request by yodax**
This upgrades nextcloud to version 20. This also solves a lot of sync client issues. 

Several notes:

- **The next version of nextcloud (21) will require php 7.3!**
- This required an extra upgrade step to add a missing column once migrated to version 20
- When migrating to version 18, there's an issue that needs a manual database change, see: https://github.com/nextcloud/server/issues/23174
- All the apps now have maximum support versions. (See attached screenshots) The upgrade takes this into account.
- Two extra php packages were required.

This works on **my** machine. Testing is greatly appreciated. I don't have any issues if someone wants to take this PR as a base and improve upon it! (glady :P) 

This doesn't show any issues in the online server status. Calendar and contacts are working

Screenshots of maximum versions that are supported: 
https://apps.nextcloud.com/apps/contacts
https://apps.nextcloud.com/apps/user_external
https://apps.nextcloud.com/apps/calendar

firefox_uxVkKjUooF
firefox_WSkTfEztdx

Upgrade log:
<details>
<p>
```
Installing Nextcloud (contacts/calendar)...
Upgrading Nextcloud --- backing up existing installation, configuration, and database to directory to /home/user-data/owncloud-backup/2020-10-23-08:25:32...

Upgrading to Nextcloud version 18.0.10

Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: gallery
Disabled incompatible app: previewgenerator
Updating <accessibility> ...
Migrate old user config

 Done
    1/1 [============================] 100%
Updated <accessibility> to 1.4.0
Updating <contacts> ...
Updated <contacts> to 3.4.1
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.8.0
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 1.7.0
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.15.2
Updating <files_videoplayer> ...
Updated <files_videoplayer> to 1.7.0
Updating <logreader> ...
Updated <logreader> to 2.3.0
Updating <password_policy> ...
Updated <password_policy> to 1.8.0
Updating <privacy> ...
Updated <privacy> to 1.2.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.8.0
Updating <recommendations> ...
Updated <recommendations> to 0.6.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.8.0
Updating <support> ...
Updated <support> to 1.1.1
Updating <survey_client> ...
Updated <survey_client> to 1.6.0
Updating <text> ...
Updated <text> to 2.0.0
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.7.0
Updating <updatenotification> ...
Updated <updatenotification> to 1.8.0
Updating <federation> ...
Updated <federation> to 1.8.0
Updating <lookup_server_connector> ...
Updated <lookup_server_connector> to 1.6.0
Updating <oauth2> ...
Updated <oauth2> to 1.6.0
Updating <user_external> ...
Updated <user_external> to 1.0.0
Updating <files> ...
Updated <files> to 1.13.1
Updating <activity> ...
Updated <activity> to 2.11.1
Updating <cloud_federation_api> ...
Updated <cloud_federation_api> to 1.1.0
Updating <dav> ...
Fix broken values of calendar objects
 Done
    0/0 [>---------------------------]   0%
Updated <dav> to 1.14.0
Updating <files_external> ...
Updated <files_external> to 1.9.0
Updating <files_sharing> ...
Updated <files_sharing> to 1.10.1
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.8.0
Updating <files_versions> ...
Updated <files_versions> to 1.11.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.8.0
Updating <workflowengine> ...
Updated <workflowengine> to 2.0.0
Updating <comments> ...
Updated <comments> to 1.8.0
Updating <nextcloud_announcements> ...
Updated <nextcloud_announcements> to 1.7.0
Updating <notifications> ...
Updated <notifications> to 2.6.0
Updating <systemtags> ...
Updated <systemtags> to 1.8.0
Updating <theming> ...
Updated <theming> to 1.9.0
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore
Checking for update of app calendar in appstore
Update app calendar from appstore
Checked for update of app "calendar" in appstore
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore
Checking for update of app contacts in appstore
Checked for update of app "contacts" in appstore
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore
Checking for update of app federation in appstore
Checked for update of app "federation" in appstore
Checking for update of app files in appstore
Checked for update of app "files" in appstore
Checking for update of app files_external in appstore
Checked for update of app "files_external" in appstore
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore
Checking for update of app files_rightclick in appstore
Checked for update of app "files_rightclick" in appstore
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore
Checking for update of app nextcloud_announcements in appstore
Checked for update of app "nextcloud_announcements" in appstore
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore
Checking for update of app settings in appstore
Checked for update of app "settings" in appstore
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore
Checking for update of app support in appstore
Checked for update of app "support" in appstore
Checking for update of app survey_client in appstore
Checked for update of app "survey_client" in appstore
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore
Checking for update of app text in appstore
Checked for update of app "text" in appstore
Checking for update of app theming in appstore
Checked for update of app "theming" in appstore
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore
Checking for update of app updatenotification in appstore
Checked for update of app "updatenotification" in appstore
Checking for update of app user_external in appstore
Checked for update of app "user_external" in appstore
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore
Checking for update of app gallery in appstore
Checked for update of app "gallery" in appstore
Checking for update of app previewgenerator in appstore
Checked for update of app "previewgenerator" in appstore
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Reset log level
Check indices of the share table.
Check indices of the filecache table.
Check indices of the twofactor_providers table.
Check indices of the login_flow_v2 table.
Check indices of the whats_new table.
Check indices of the cards table.
Check indices of the cards_properties table.
Check indices of the calendarobjects_props table.
Adding calendarobject_calid_index index to the calendarobjects_props table, this can take some time...
calendarobjects_props table updated successfully.
Check indices of the schedulingobjects table.
Adding schedulobj_principuri_index index to the schedulingobjects table, this can take some time...
schedulingobjects table updated successfully.
All tables already up to date!

Upgrading to Nextcloud version 19.0.4

Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: previewgenerator
Updating <accessibility> ...
Migrate old user config

 Done
    1/1 [============================] 100%
Updated <accessibility> to 1.5.0
Updating <calendar> ...
Updated <calendar> to 2.0.3
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.9.0
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 1.8.0
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.16.0
Updating <files_videoplayer> ...
Updated <files_videoplayer> to 1.8.0
Updating <logreader> ...
Updated <logreader> to 2.4.0
Updating <password_policy> ...
Updated <password_policy> to 1.9.1
Updating <photos> ...
Updated <photos> to 1.1.0
Updating <privacy> ...
Updated <privacy> to 1.3.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.9.0
Updating <recommendations> ...
Updated <recommendations> to 0.7.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.9.0
Updating <settings> ...
Updated <settings> to 1.1.0
Updating <support> ...
Updated <support> to 1.2.1
Updating <survey_client> ...
Updated <survey_client> to 1.7.0
Updating <text> ...
Updated <text> to 3.0.1
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.8.0
Updating <updatenotification> ...
Updated <updatenotification> to 1.9.0
Updating <viewer> ...
Updated <viewer> to 1.3.0
Updating <federation> ...
Updated <federation> to 1.9.0
Updating <lookup_server_connector> ...
Updated <lookup_server_connector> to 1.7.0
Updating <oauth2> ...
Updated <oauth2> to 1.7.0
Updating <files> ...
Updated <files> to 1.14.0
Updating <activity> ...
Updated <activity> to 2.12.1
Updating <cloud_federation_api> ...
Updated <cloud_federation_api> to 1.2.0
Updating <dav> ...
Fix broken values of calendar objects
 Done
    0/0 [----->----------------------]   0%
Updated <dav> to 1.15.0
Updating <files_external> ...
Updated <files_external> to 1.10.0
Updating <files_sharing> ...
Updated <files_sharing> to 1.11.0
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.9.0
Updating <files_versions> ...
Updated <files_versions> to 1.12.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.9.0
Updating <workflowengine> ...
Updated <workflowengine> to 2.1.0
Updating <comments> ...
Updated <comments> to 1.9.0
Updating <nextcloud_announcements> ...
Updated <nextcloud_announcements> to 1.8.0
Updating <notifications> ...
Updated <notifications> to 2.7.0
Updating <systemtags> ...
Updated <systemtags> to 1.9.0
Updating <theming> ...
Updated <theming> to 1.10.0
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore
Checking for update of app calendar in appstore
Update app calendar from appstore
Checked for update of app "calendar" in appstore
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore
Checking for update of app contacts in appstore
Checked for update of app "contacts" in appstore
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore
Checking for update of app federation in appstore
Checked for update of app "federation" in appstore
Checking for update of app files in appstore
Checked for update of app "files" in appstore
Checking for update of app files_external in appstore
Checked for update of app "files_external" in appstore
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore
Checking for update of app files_rightclick in appstore
Checked for update of app "files_rightclick" in appstore
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore
Checking for update of app nextcloud_announcements in appstore
Checked for update of app "nextcloud_announcements" in appstore
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore
Checking for update of app photos in appstore
Checked for update of app "photos" in appstore
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore
Checking for update of app settings in appstore
Checked for update of app "settings" in appstore
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore
Checking for update of app support in appstore
Checked for update of app "support" in appstore
Checking for update of app survey_client in appstore
Checked for update of app "survey_client" in appstore
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore
Checking for update of app text in appstore
Checked for update of app "text" in appstore
Checking for update of app theming in appstore
Checked for update of app "theming" in appstore
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore
Checking for update of app updatenotification in appstore
Checked for update of app "updatenotification" in appstore
Checking for update of app user_external in appstore
Checked for update of app "user_external" in appstore
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore
Checking for update of app previewgenerator in appstore
Checked for update of app "previewgenerator" in appstore
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Reset log level
Check indices of the share table.
Check indices of the filecache table.
Check indices of the twofactor_providers table.
Check indices of the login_flow_v2 table.
Check indices of the whats_new table.
Check indices of the cards table.
Check indices of the cards_properties table.
Check indices of the calendarobjects_props table.
Check indices of the schedulingobjects table.
Check indices of the oc_properties table.
Adding properties_path_index index to the oc_properties table, this can take some time...
oc_properties table updated successfully.
All tables already up to date!

Upgrading to Nextcloud version 20.0.0

Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Setting log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: previewgenerator
Updating <accessibility> ...
Migrate old user config

 Done
    1/1 [============================] 100%
Updated <accessibility> to 1.6.0
Updating <calendar> ...
Updated <calendar> to 2.1.2
Updating <contactsinteraction> ...
Updated <contactsinteraction> to 1.1.0
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.10.1
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 2.0.1
Updating <files_rightclick> ...
Updated <files_rightclick> to 0.17.0
Updating <files_videoplayer> ...
Updated <files_videoplayer> to 1.9.0
Updating <logreader> ...
Updated <logreader> to 2.5.0
Updating <photos> ...
Updated <photos> to 1.2.0
Updating <privacy> ...
Updated <privacy> to 1.4.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.10.0
Updating <recommendations> ...
Updated <recommendations> to 0.8.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.10.0
Updating <settings> ...
Updated <settings> to 1.2.0
Updating <support> ...
Updated <support> to 1.3.0
Updating <survey_client> ...
Updated <survey_client> to 1.8.0
Updating <text> ...
Updated <text> to 3.1.0
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.9.0
Updating <updatenotification> ...
Updated <updatenotification> to 1.10.0
Updating <viewer> ...
Updated <viewer> to 1.4.0
Updating <federation> ...
Updated <federation> to 1.10.1
Updating <lookup_server_connector> ...
Updated <lookup_server_connector> to 1.8.0
Updating <oauth2> ...
Updated <oauth2> to 1.8.0
Updating <password_policy> ...
Updated <password_policy> to 1.10.1
Updating <files> ...
Updated <files> to 1.15.0
Updating <activity> ...
Updated <activity> to 2.13.1
Updating <cloud_federation_api> ...
Updated <cloud_federation_api> to 1.3.0
Updating <dav> ...
Fix broken values of calendar objects
 Done
    0/0 [----->----------------------]   0%
Updated <dav> to 1.16.0
Updating <files_external> ...
Updated <files_external> to 1.11.1
Updating <files_sharing> ...
Updated <files_sharing> to 1.12.0
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.10.1
Updating <files_versions> ...
Updated <files_versions> to 1.13.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.10.0
Updating <workflowengine> ...
Updated <workflowengine> to 2.2.0
Updating <comments> ...
Updated <comments> to 1.10.0
Updating <nextcloud_announcements> ...
Updated <nextcloud_announcements> to 1.9.0
Updating <notifications> ...
Updated <notifications> to 2.8.0
Updating <systemtags> ...
Updated <systemtags> to 1.10.0
Updating <theming> ...
Updated <theming> to 1.11.0
Checking for update of app accessibility in appstore
Checked for update of app "accessibility" in appstore
Checking for update of app activity in appstore
Checked for update of app "activity" in appstore
Checking for update of app calendar in appstore
Checked for update of app "calendar" in appstore
Checking for update of app cloud_federation_api in appstore
Checked for update of app "cloud_federation_api" in appstore
Checking for update of app comments in appstore
Checked for update of app "comments" in appstore
Checking for update of app contacts in appstore
Checked for update of app "contacts" in appstore
Checking for update of app contactsinteraction in appstore
Checked for update of app "contactsinteraction" in appstore
Checking for update of app dav in appstore
Checked for update of app "dav" in appstore
Checking for update of app federatedfilesharing in appstore
Checked for update of app "federatedfilesharing" in appstore
Checking for update of app federation in appstore
Checked for update of app "federation" in appstore
Checking for update of app files in appstore
Checked for update of app "files" in appstore
Checking for update of app files_external in appstore
Checked for update of app "files_external" in appstore
Checking for update of app files_pdfviewer in appstore
Checked for update of app "files_pdfviewer" in appstore
Checking for update of app files_rightclick in appstore
Checked for update of app "files_rightclick" in appstore
Checking for update of app files_sharing in appstore
Checked for update of app "files_sharing" in appstore
Checking for update of app files_trashbin in appstore
Checked for update of app "files_trashbin" in appstore
Checking for update of app files_versions in appstore
Checked for update of app "files_versions" in appstore
Checking for update of app files_videoplayer in appstore
Checked for update of app "files_videoplayer" in appstore
Checking for update of app logreader in appstore
Checked for update of app "logreader" in appstore
Checking for update of app lookup_server_connector in appstore
Checked for update of app "lookup_server_connector" in appstore
Checking for update of app nextcloud_announcements in appstore
Checked for update of app "nextcloud_announcements" in appstore
Checking for update of app notifications in appstore
Checked for update of app "notifications" in appstore
Checking for update of app oauth2 in appstore
Checked for update of app "oauth2" in appstore
Checking for update of app password_policy in appstore
Checked for update of app "password_policy" in appstore
Checking for update of app photos in appstore
Checked for update of app "photos" in appstore
Checking for update of app privacy in appstore
Checked for update of app "privacy" in appstore
Checking for update of app provisioning_api in appstore
Checked for update of app "provisioning_api" in appstore
Checking for update of app recommendations in appstore
Checked for update of app "recommendations" in appstore
Checking for update of app serverinfo in appstore
Checked for update of app "serverinfo" in appstore
Checking for update of app settings in appstore
Checked for update of app "settings" in appstore
Checking for update of app sharebymail in appstore
Checked for update of app "sharebymail" in appstore
Checking for update of app support in appstore
Checked for update of app "support" in appstore
Checking for update of app survey_client in appstore
Checked for update of app "survey_client" in appstore
Checking for update of app systemtags in appstore
Checked for update of app "systemtags" in appstore
Checking for update of app text in appstore
Checked for update of app "text" in appstore
Checking for update of app theming in appstore
Checked for update of app "theming" in appstore
Checking for update of app twofactor_backupcodes in appstore
Checked for update of app "twofactor_backupcodes" in appstore
Checking for update of app updatenotification in appstore
Checked for update of app "updatenotification" in appstore
Checking for update of app user_external in appstore
Checked for update of app "user_external" in appstore
Checking for update of app viewer in appstore
Checked for update of app "viewer" in appstore
Checking for update of app workflowengine in appstore
Checked for update of app "workflowengine" in appstore
Checking for update of app previewgenerator in appstore
Checked for update of app "previewgenerator" in appstore
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Resetting log level
Check indices of the share table.
Check indices of the filecache table.
Check indices of the twofactor_providers table.
Check indices of the login_flow_v2 table.
Check indices of the whats_new table.
Check indices of the cards table.
Check indices of the cards_properties table.
Check indices of the calendarobjects_props table.
Check indices of the schedulingobjects table.
Check indices of the oc_properties table.
Done.
All tables already up to date!
Nextcloud is already latest version
```
</p>
</details>

### CODE REVIEW

**Comment by Synchro** [comment]
FWIW, upgrading PHP should be a no-brainer. [PHP 7.2 becomes completely unsupported in 1 month's time](https://www.php.net/supported-versions) and 7.3 loses active support at the same time. PHP 8 will be out next month too. We should be going straight to 7.4 ASAP, assuming that NC 20 supports that.

**Comment by yodax** [comment]
Upgrading php needs to take into consideration that we support upgrades from version 14 onwards. That version probably doesn’t support the new php version (we should check that). 

I wouldn’t mind stopping that, but it would greatly complicate upgrades for people.

**Comment by Synchro** [comment]
I found the answer to that. [Nextcloud 14 supports a maximum of PHP 7.2](https://docs.nextcloud.com/server/14/admin_manual/installation/system_requirements.html), [NC 15-17 support 7.3](https://docs.nextcloud.com/server/15/admin_manual/installation/system_requirements.html), and [18 works on 7.4](https://docs.nextcloud.com/server/18/admin_manual/installation/system_requirements.html), so you would indeed need to do a little hopscotch to sequence upgrading both NC and PHP, but we would need to do some of that anyway if we're not going to end up running a completely obsolete PHP version.

File: setup/nextcloud.sh [hunk 1]
```diff
@@ -91,8 +97,14 @@ InstallNextcloud() {
...
 # Nextcloud Version to install. Checks are done down below to step through intermediate versions.
-nextcloud_ver=17.0.6
-nextcloud_hash=50b98d2c2f18510b9530e558ced9ab51eb4f11b0
+nextcloud_ver=20.0.1
+nextcloud_hash=f2b3faa570c541df73f209e873a1c2852e79eab8
+contacts_ver=3.4.1
+contacts_hash=aee680a75e95f26d9285efd3c1e25cf7f3bfd27e
+calendar_ver=2.1.2
+calendar_hash=930c07863bb7a65652dec34793802c8d80502336
+user_external_var=1.0.0
```
Comment (jvolkenant) [inline setup/nextcloud.sh:106]: Should this be user_external_ver?

File: setup/nextcloud.sh [hunk 2]
```diff
@@ -141,23 +153,39 @@ if [ ! -d /usr/local/lib/owncloud/ ] || [[ ! ${CURRENT_NEXTCLOUD_VER} =~ ^$nextc
...
+                        CURRENT_NEXTCLOUD_VER="18.0.10"
+	        fi
+	        if [[ ${CURRENT_NEXTCLOUD_VER} =~ ^18 ]]; then
+                        InstallNextcloud 19.0.4 01e98791ba12f4860d3d4047b9803f97a1b55c60 3.4.1 aee680a75e95f26d9285efd3c1e25cf7f3bfd27e 2.0.3 9d9717b29337613b72c74e9914c69b74b346c466 1.0.0 3bf2609061d7214e7f0f69dd8883e55c4ec8f50a
+                        CURRENT_NEXTCLOUD_VER="19.0.4"
+	        fi
 	fi
 
-	InstallNextcloud $nextcloud_ver $nextcloud_hash
+	InstallNextcloud $nextcloud_ver $nextcloud_hash $contacts_ver $contacts_hash $calendar_ver $calendar_hash $user_external_var $user_external_hash
```
Comment (jvolkenant) [inline setup/nextcloud.sh:185]: Should this be $user_external_ver?

**Review by jvolkenant — COMMENTED** [review]
Looks good. I'll have to pull and give this a shot

File: setup/nextcloud.sh (hunk 1 above)
Comment (yodax) [reply setup/nextcloud.sh]: Yes 👍

**Comment by jvolkenant** [comment]
Upgraded to 20.0.1 successfully.  Not sure that I'll be able to test a fresh install of whatever tag installs Nextcloud 13 and see if it goes to 20 as another test.

**Comment by JoshData** [comment]
Excellent work here. Has anyone reviewed whether any changes are necessary in the nginx config, the Nextcloud cron job, etc?

**Comment by yodax** [comment]
> Excellent work here. Has anyone reviewed whether any changes are necessary in the nginx config, the Nextcloud cron job, etc?

I couldn't find anything here:

https://docs.nextcloud.com/server/20/developer_manual/app_publishing_maintenance/upgrade-guide.html#id9
https://github.com/nextcloud/server/issues/20953
https://github.com/nextcloud/server/issues/17131
https://github.com/nextcloud/server/issues/20953

The status checks from nextcloud pass except for the usual sqlite remark.

**Comment by JoshData** [comment]
Cool cool. I'll give this PR a try on my box then.

**Comment by dumblob** [comment]
>Has anyone reviewed whether any changes are necessary in the nginx config, the Nextcloud cron job, etc?

Does the NC cron job run every 5 minutes? It's the "new" recommended setting as per NC documentation.

**Comment by yodax** [comment]
https://github.com/mail-in-a-box/mailinabox/blob/8b166f30415611c6f3c846065a1b9a67c5d12de2/setup/nextcloud.sh#L316

It already runs cron every 5 minutes.

**Comment by JoshData** [comment]
Worked great for me. Merging!

**Comment by stateless** [comment]
Is there an accepted way to upgrade to HEAD in order to test before release?

**Comment by cdp1337** [comment]
Excellent, I just recently stumbled onto this project and only noticed nextcloud was a bit outdated when I installed file-link for Thunderbird.  Looking forward to the next release that'll contain this!  (Kolab has left a bad taste in my mouth for running email on master.)

**Comment by mccompunerd** [comment]
I am also interested in doing this upgrade ahead of the MIAB upgrade. I am fairly tech savvy, but don't want to play with my box too much unless I have rough directions. I really don't want to tinker too much with it because it hosts my non-Google e-mail and shared calendars.

**Pull request merged**